### PR TITLE
Use variable for bom version

### PR DESCRIPTION
**Pull request by yiliuTo**
In dependency pattern of bom libraries, use variable to display version number.

### CODE REVIEW

**Comment by saragluna** [comment]
If it's still in progress, how about converting it to draft?

File: sdk/spring/spring-reference.yml [hunk 1]
```diff
@@ -260,14 +301,15 @@
             github: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/spring/azure-spring-cloud-stream-binder-servicebus-topic
           springProperties:
             compatibilityRange: "[2.2.11.RELEASE,2.4.0-M1)"
```
Comment (backwind1233) [inline sdk/spring/spring-reference.yml:303]: compatibilityRange after bom

File: sdk/spring/spring-reference.yml [hunk 2]
```diff
@@ -148,7 +188,7 @@
             github: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-jca
         - artifactId: azure-spring-boot-starter-keyvault-certificates
           groupId: com.azure.spring
-          versionPreview: 3.0.0-beta.4
+          versionPreview: 3.0.0-beta.6
```
Comment (backwind1233) [inline sdk/spring/spring-reference.yml:191]: not beta.5,right?

File: sdk/spring/spring-reference.yml [hunk 3]
```diff
@@ -51,6 +51,46 @@
...
+                </dependencies>
+              </dependencyManagement>
+            gradle: |-
+              dependencyManagement {
+                imports {
+                  mavenBom "com.azure.spring:azure-spring-cloud-dependencies:${azure.version}"
+                }
+              }
+          springProperties:
+            starter: false
```
Comment (backwind1233) [inline sdk/spring/spring-reference.yml:87]: we should add `starter` property here only or for every entry

File: sdk/spring/spring-reference.yml [hunk 4]
```diff
@@ -51,6 +51,46 @@
...
+            compatibilityRange: "[2.2.11.RELEASE,2.4.0-M1)"
+            mappings:
+              - compatibilityRange: "[2.0.0.RELEASE,2.3.0.M1)"
+                groupId: com.microsoft.azure
+                artifactId: spring-cloud-azure-dependencies
```
Comment (backwind1233) [inline sdk/spring/spring-reference.yml:92]: what's the difference between spring-cloud-azure-dependencies and azure-spring-cloud-dependencies?

File: sdk/spring/spring-reference.yml (hunk 2 above)
Comment (yiliuTo) [reply sdk/spring/spring-reference.yml:191]: Correct, our latest version is 6

File: sdk/spring/spring-reference.yml (hunk 3 above)
Comment (yiliuTo) [reply sdk/spring/spring-reference.yml:87]: the default value is false, so it's ok to add this property only if it's true.

File: sdk/spring/spring-reference.yml (hunk 4 above)
Comment (yiliuTo) [reply sdk/spring/spring-reference.yml:92]: The mapping property is used for version mapping with Spring Boot. So it's necessary for us spring libraries as our libs cannot support all Spring Boot versions, I think.

**Review by backwind1233 — APPROVED** [review]
LGTM

**Pull request merged**